app/src/main/java/org/mozilla/focus/browser/FullscreenCallbacks.kt

```diff
@@ -16,18 +16,26 @@ import kotlinx.android.synthetic.main.fragment_browser.*
 import org.mozilla.focus.R
 import org.mozilla.focus.iwebview.IWebView
 import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.widget.OnInterceptTouchEventFrameLayout
 
 /**
  * An [IWebView.Callback] that only handles the fullscreen related callbacks.
  */
-class FullscreenCallbacks(
-        private val browserFragment: BrowserFragment
+open class FullscreenCallbacks(
+    private val browserFragment: BrowserFragment,
+    private val telemetryWrapper: TelemetryWrapper = TelemetryWrapper
```

Comment:
Do we need to do this elsewhere for mocking, in tests? Is this something we need to add to avoid tests sending telemetry...?

Reply:
In theory, anywhere we need to mock telemetry, we need to pass in the dependency so yes something like this. However, it's probably not worth refactoring all the code to pass in the telemetry dependency unless it's strictly needed.

wrt testing, I don't think manually passing in the telemetry dependency is a sustainable way to avoid sending tests during telemetry. I think we need a comprehensive rethink of our approaches. There's:
- robolectric: https://github.com/mozilla-mobile/firefox-tv/issues/1300 (which improbably sends events)
- A discussion thread: https://groups.google.com/a/mozilla.com/d/msg/android-product-team/BJADnmwk2V4/NOanXwIECgAJ
- This components issue: https://github.com/mozilla-mobile/android-components/issues/695
- This fenix issue: https://github.com/mozilla-mobile/fenix/issues/124
- An issue I just filed against glean: https://bugzilla.mozilla.org/show_bug.cgi?id=1514336

I'll bring it up in the original thread.